test-infra/auto-deploy/Dockerfile

```diff
@@ -0,0 +1,165 @@
+    chmod 700 /tmp/get_helm.sh && \
+    /tmp/get_helm.sh && \
+    rm /tmp/get_helm.sh
+
+# Initialize helm
+RUN helm init --client-only
+
+# Install  Node.js
+RUN curl -sL https://deb.nodesource.com/setup_8.x | bash - \
+    && apt-get install -y nodejs
+
+# Install yarn
+RUN curl -sS http://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+    && echo "deb http://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+    && apt-get update -yqq \
+    && apt-get install -yqq --no-install-recommends yarn
+
+# Install glide
+RUN cd /tmp && \
+    wget -O glide-v0.13.0-linux-amd64.tar.gz \
+    https://github.com/Masterminds/glide/releases/download/v0.13.0/glide-v0.13.0-linux-amd64.tar.gz && \
+    tar -xvf glide-v0.13.0-linux-amd64.tar.gz && \
+    mv ./linux-amd64/glide /usr/local/bin/
+
+# Install ksonnet. We install multiple versions of ks to support different versions
+# of ksonnet applications. Newer versions of ksonnet are backwards compatible but
+# that can require upgrading the app which isn't something we want to be forced to.
+# (see https://github.com/kubeflow/testing/issues/220).
+RUN cd /tmp && \
+    wget -O ks.tar.gz \
+    https://github.com/ksonnet/ksonnet/releases/download/v0.11.0/ks_0.11.0_linux_amd64.tar.gz && \
+    tar -xvf ks.tar.gz && \
+    mv ks_0.11.0_linux_amd64/ks /usr/local/bin && \
+    chmod a+x /usr/local/bin/ks
+
+RUN cd /tmp && \
+    wget -O ks-12.tar.gz \
+    https://github.com/ksonnet/ksonnet/releases/download/v0.12.0/ks_0.12.0_linux_amd64.tar.gz && \
+    tar -xvf ks-12.tar.gz && \
+    mv ks_0.12.0_linux_amd64/ks /usr/local/bin/ks-12 && \
+    chmod a+x /usr/local/bin/ks-12
+
+RUN cd /tmp && \
+    wget -O ks-13.tar.gz \
+    https://github.com/ksonnet/ksonnet/releases/download/v0.13.1/ks_0.13.1_linux_amd64.tar.gz && \
+    tar -xvf ks-13.tar.gz && \
+    mv ks_0.13.1_linux_amd64/ks /usr/local/bin/ks-13 && \
+    chmod a+x /usr/local/bin/ks-13
+
+RUN cd /tmp && \
+    wget https://github.com/google/jsonnet/archive/v0.11.2.tar.gz && \
+    tar -xvf v0.11.2.tar.gz && \
+    cd jsonnet-0.11.2 && \
+    make && \
+    mv jsonnet /usr/local/bin && \
+    rm -rf /tmp/v0.11.2.tar.gz && \
+    rm -rf /tmp/jsonnet-0.11.2
+
+# Install various python libraries for both Python 2 and 3 (for now)
+# Don't upgrade pip for now because it seems to be broken
+# https://github.com/pypa/pip/issues/5240
+COPY ./Pipfile ./Pipfile.lock /tmp/
+
+RUN cd /tmp/ && \
+    pip2 install -U wheel filelock && \
+    pip2 install pipenv && \
+    pip2 install requests && \
+    pip2 install google-api-python-client && \
+    pip2 install prometheus_client && \
+    pipenv install --system --two && \
+    pip3 install -U wheel filelock
+
+RUN pip3 install pipenv==2018.10.9
+RUN cd /tmp/ && pipenv install --system --three
+
+RUN pip install yq
+
+COPY checkout.sh /usr/local/bin
+RUN chmod a+x /usr/local/bin/checkout.sh
+
+COPY workflows.sh /usr/local/bin
+RUN chmod a+x /usr/local/bin/workflows.sh
```

Comment:
What is this code that's being backed into the docker image? Why not checkout a fresh copy of the code and then invoke the script inside the repo?